### PR TITLE
clamtk: init at 6.16

### LINKED ISSUE #263331: Package request: ClamTk
**Project description**

Easy to use, lightweight front-end for ClamAV(Clam Antivirus)

**Metadata**

* homepage URL: https://github.com/dave-theunsub/clamtk
* source URL: https://github.com/dave-theunsub/clamtk
* license: All rights reserved.  This program is free software; you can redistribute it and/or modify it under the terms of either: "GPL1+" or "Artistic License"
* platforms: linux

### PR DESCRIPTION
**Pull request by chewblacka**
## Description of changes
clamtk is a frontend for ClamAV (Clam Antivirus)
https://github.com/dave-theunsub/clamtk

Closes #263331

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/cl/clamtk/package.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  buildInputs = [ perl clamav ];
+  propagatedBuildInputs = with perlPackages; [ Glib LWP LWPProtocolHttps TextCSV JSON LocaleGettext Gtk3 ];
+
+  preConfigure = "touch Makefile.PL";
+  doCheck = false;
+  outputs = [ "out" "man" ];
+
+  postPatch = ''
+    # Set the correct paths for clamav programs
+    sed -i lib/App.pm -e  "93s@.*@        : \'${clamavpath}/freshclam\';@"
```
Comment (peterhoeg) [inline pkgs/by-name/cl/clamtk/package.nix:33]: This is a bit iffy. I'd much prefer something like:
```sh
substituteInPlace lib/App.pm \
  --replace /usr/bin/freshclam ${lib.getBin clamav}/bin/freshclam \
  --replace /usr/bin/sigtool ${lib.getBin clamav}/bin/sigtool
```
and so on for the others

If you are dependent on specific line numbers, it gets really painful to maintain

File: pkgs/by-name/cl/clamtk/package.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    sed -i lib/App.pm -e "114s@.*@        : \'${clamavpath}/sigtool\';@"
+    sed -i lib/App.pm -e "123s@.*@        : \'${clamavpath}/clamscan\';@"
+    sed -i lib/App.pm -e "134s@.*@        : \'${clamavpath}/clamdscan\';@"
+
+    # Icons path
+    sed -i lib/App.pm -e "s@path->{ images } =.*@path->{ images } = '$out/share/pixmaps';@"
```
Comment (peterhoeg) [inline pkgs/by-name/cl/clamtk/package.nix:39]: Same as above. Just substitute the path. `/usr/share/pixmaps` to `$out/share/pixmaps`.

File: pkgs/by-name/cl/clamtk/package.nix (hunk 1 above)
Comment (chewblacka) [reply pkgs/by-name/cl/clamtk/package.nix]: Thanks! Your method is much better. Have replaced `sed` with `substituteInPlace` for all occurrences in `App.pm` as suggested. Also for the icons path. And also managed to do it for `crontab`. Much appreciated as always. Many thanks!

**Comment by peterhoeg** [comment]
last thing: please put in a comment concerning *why* you are disabling checks. Even if it's just because there aren't any.

I haven't tried this application (and am not likely to do so), but otherwise looks to go when the comment above is addressed.

**Pull request merged**